src/monome.js

```diff
@@ -1,35 +1,62 @@
 import {
+  log,
   err,
   ERR_WRITE,
   ERR_WRITE_MISSING_BYTES,
   USB_XFER_STATUS_OK,
 } from './utils.js';
 
+import { getInterfaceForVendor } from './webmonome.js';
+
 export default class Monome {
   constructor(device) {
     this.device = device;
     this.callbacks = {};
+
+    // set i/o based on chosen interface endpoints
+    // TODO: cleanup :'(
+    try {
+      this.device.configuration.interfaces[
+        getInterfaceForVendor(device.vendorId)
+      ].alternates[0].endpoints.forEach(({ direction, endpointNumber }) => {
+        if (direction === 'in' && this.endpointIn === undefined) {
+          this.endpointIn = endpointNumber;
+        } else if (direction === 'out' && this.endpointOut === undefined) {
+          this.endpointOut = endpointNumber;
+        }
+      });
+    } catch (e) {
+      log(e, 2);
+    }
+
+    // if unabled to dynmically set, we can try semi-sensible defaults
+    this.endpointIn = this.endpointIn === undefined ? 1 : this.endpointIn;
+    this.endpointOut = this.endpointOut === undefined ? 2 : this.endpointOut;
   }
 
   async listen() {
     if (!this.isConnected) return;
     const result = await this.read(64);
 
-    if (result.status === 'ok' && result.data.byteLength > 2) {
-      this.processData(result.data, 2);
+    // TODO: did i have a reason for skipping the first two bytes here?
```

Comment:
Great! I'm not sure either; I just always assumed it was correct. Not skipping works fine on my Series device 🎉